scripts/patterns-reorg/models.py

```diff
@@ -124,20 +133,15 @@ def determine_redirects(
         if isinstance(entry, Entry):
             result.update(determine_redirects(entry.children))
     
-        if entry.slug is None or not entry.from_file:
-            continue
-        
-        old_url = str(PurePath(entry.from_file).with_suffix(""))
-        redirect_to = f"{prefix}{entry.slug.removeprefix('/')}"
-        result[old_url] = redirect_to
-
-        # We need to add two links for each index entry because we can
-        # have two links possible. For example, `/run/index` and `/run`
-        # point to the same page.
-        old_folder, old_file_name = old_url.split('/')
-        if old_file_name == "index":
-            result[f"{old_folder}/"] = redirect_to
-
-
+            if entry.slug is None or not entry.from_file:
+                continue
+            
+            old_url = str(PurePath(entry.from_file).with_suffix(""))
+            redirect_to = f"{prefix}{entry.slug.removeprefix('/')}"
+            add_redirect_to_dict(result, old_url, redirect_to)
+
+        elif isinstance(entry, DeletedPage):    
```

Comment:
I used an `elif` and not a `continue` in the first case because I think it's more readable.